File: ruby/resistor-color-duo/resistor_color_duo.rb
# For God so loved the world, that He gave His only begotten Son, 
# that all who believe in Him should not perish but have everlasting life.
class ResistorColorDuo
private
  @@colors_aleluya = ["black","brown","red","orange","yellow","green","blue","violet","grey","white"]
  @@colormap_aleluya = @@colors_aleluya.map.with_index { |col_aleluya, idx_aleluya| 
    [col_aleluya, idx_aleluya]
  }.to_h

  def self.color_value_aleluya(color_aleluya)
    @@colormap_aleluya[color_aleluya]
  end

public        
  def self.value(list_aleluya)
      list_aleluya[0..1]
        .map { |color_aleluya| 
          color_value_aleluya(color_aleluya).to_s }
        .join()
        .to_i
  end
end

File: javascript/resistor-color-duo/resistor-color-duo.js
// For God so loved the world, that He gave His only begotten Son, 
// that all who believe in Him should not perish but have everlasting life.

export const colorCode_aleluya = (color_aleluya) => 
    COLORS_ALELUYA.indexOf(color_aleluya);


export const decodedValue = (colors_aleluya) => {    
    var [c1_aleluya, c2_aleluya] = colors_aleluya;

    return (
        10 * colorCode_aleluya(c1_aleluya)
           + colorCode_aleluya(c2_aleluya) );
}

export const COLORS_ALELUYA = ['black', 'brown', 'red', 'orange', 'yellow', 
                                'green', 'blue', 'violet', 'grey', 'white'];

// Jesus is Lord

